src/stdlib_stats_median.fypp

```diff
@@ -43,16 +41,18 @@ contains
 
         x_tmp = reshape(x, [n])
 
-        call sort(x_tmp)
+        call select(x_tmp, c, val)
 
         if (mod(n, 2_int64) == 0) then
+          call select(x_tmp, c+1, val1, left = c)
```

Comment:
I am wondering if 
```fortran
val1 = minval(x_tmp(c+1:n))
```
would be more efficient than
```fortran
call select(x_tmp, c+1, val1, left = c)
```
, since `all(x_tmp(c+1:n)  > x_tmp(c))`?

Reply:
Yes I anticipate `minval` should be faster will only require one scan of the data.

Reply:
Done. I replaced the second `select` by `minval`